substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/config/SerializationConfigurationType.java

```diff
@@ -55,7 +56,14 @@ public SerializationConfigurationType(ConfigurationCondition condition, String q
     public void printJson(JsonWriter writer) throws IOException {
         writer.append('{').indent().newline();
         ConfigurationConditionPrintable.printConditionAttribute(condition, writer);
-        writer.quote(SerializationConfigurationParser.NAME_KEY).append(':').quote(qualifiedJavaName);
+
+        if (qualifiedJavaName.contains("$$Lambda$") && !qualifiedJavaName.contains("/")) {
```

Comment:
I think abusing qualifiedJavaName is not necessary here. It can also lead to duplicates during writing. We need to put the new value properly in the model and then change the equality function to compare that as well. 

Also, we need assertions that both can't be populated.

Reply:
New attribute `isLambdaCapturingClass` added to distinct lambda capturing classes from classes that should be registered for serialization.
Assertion added as suggested.